### PR TITLE
feat(vacation-calendar): sort users in calendar view

### PR DESCRIPTION
**Pull request by fridajac**
### Context

Users are now sorted alphabetically on the frontend when loaded, providing a consistent and predictable user experience.

Also added comprehensive tests covering sorting, edge cases, and highlight logic

### Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have verified that the code builds perfectly fine on my local system
- [x] I have verified that my code follows the style already available in the repository
- [x] A changeset describing the change and affected packages. ([more info](https://github.com/AxisCommunications/backstage-plugins/blob/main/CONTRIBUTING.md#changesets))
- [x] Tests for new functionality and regression tests for bug fixes

### CODE REVIEW

File: .changeset/quick-candles-unite.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@axis-backstage/plugin-jira-dashboard-backend': patch
+---
+
+Refactored sequential for loops in the Jira Dashboard backend to use Promise.all / Promise.allSettled, reducing latency when multiple projects or filters are fetched.
```
Comment (anicke) [inline .changeset/quick-candles-unite.md]: Is this about sorting? The changelog makes no sense

**Review by anicke — CHANGES_REQUESTED** [review]

File: .changeset/quick-candles-unite.md (hunk 1 above)
Comment (fridajac) [reply .changeset/quick-candles-unite.md]: Oh no I accidentally replaced mine with some old one. I fix.

**Review by anicke — APPROVED** [review]
